### ISSUE TITLE
Provide a flag to open data packages with not safe paths?

### ISSUE THREAD

**Issue by vitorbaptista** [frictionlessdata/datapackage-py#171]
# Overview

We could allow to open data package with not safe paths if user explicitly choose it:

```python
package = Package(descriptor, safe=False)
```

---

_From @vitorbaptista_ 

This is what I'm trying to do:

```python
import datapackage

descriptor = {
    'name': 'my-datapackage',
    'resources': [
      {
        'name': 'data',
        'path': '/tmp/my-data.csv'
      }
    ]
}
dp = datapackage.Package(descriptor)
# *** tableschema.exceptions.DataPackageException: Local path "/tmp/my-data.csv" is not safe
```

Reading the `helpers.is_safe_path()` method:

https://github.com/frictionlessdata/datapackage-py/blob/3d25cfd2013c0ddb6b5f9ceb7724dcd111fc9905/datapackage/helpers.py#L189-L196

Makes me think that what I'm trying to do isn't possible anymore. Any known workarounds?

(cc @roll)

**Comment by roll** [comment]
@vitorbaptista 
There was a long discussion on path security in Specs Working Group that has been finished with this resolution (http://specs.frictionlessdata.io/data-resource/#path-data-in-files):
> POSIX paths (unix-style with / as separator) are supported for referencing local files, with the security restraint that they MUST be relative siblings or children of the descriptor. Absolute paths (/) and relative parent paths (../) MUST NOT be used, and implementations SHOULD NOT support these path types.

> / (absolute path) and ../ (relative parent path) are forbidden to avoid security vulnerabilities when implementing data package tools. These limitations on resource path ensure that resource paths only point to files within the data package directory and its subdirectories. This prevents data package tools being exploited by a malicious user to gain unintended access to sensitive information. For example, suppose a data package hosting service stores packages on disk and allows access via an API. A malicious user uploads a data package with a resource path like /etc/passwd. The user then requests the data for that resource and the server naively opens /etc/passwd and returns that data to the caller.

So for now a valid data package descriptor can't have path `/tmp/my-data.csv`. And I think it's really good - to protect all users by default. But as an implementation we could provide an additional functionality like a flag to support unsafe paths. Something like this:

```python
package = Package(descriptor, safe=False)
```
And disable this `is_safe_path` checks for this case.

**Comment by vitorbaptista** [comment]
@roll Taking a step back, what I want to do is create a datapackage with files from the local filesystem. Apparently, the way I can do this with the current code is by adding the resources in a folder, and passing that as a `base_path` when creating the `Package()` object. Although that makes my job a bit harder, I think it's an OK-ish compromise. I would suggest not implementing the `safe=False` flag for now if others haven't complained about it.

**Comment by vitorbaptista** [comment]
@roll So, it didn't work out. I have a datapackage that already has a local resource (it's inside a zip file), and I need to add more local resources. I tried changing the `base_path` to the new local resources' base path, but that broke the original resource the DP has.

In the end, I had to mock out `datapackage.helpers.is_safe_path`. Check https://github.com/openspending/datapackage-pipelines-fiscal/commit/20bbf6e68162cf012a42534293d97c6d7b13d3c8.
